dspace-api/src/main/java/org/dspace/app/orcid/service/impl/OrcidSynchronizationServiceImpl.java

```diff
@@ -0,0 +1,258 @@
+import org.dspace.app.orcid.service.OrcidSynchronizationService;
+import org.dspace.app.profile.OrcidEntitySyncPreference;
+import org.dspace.app.profile.OrcidProfileDisconnectionMode;
+import org.dspace.app.profile.OrcidProfileSyncPreference;
+import org.dspace.app.profile.OrcidSynchronizationMode;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@link OrcidSynchronizationService}.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class OrcidSynchronizationServiceImpl implements OrcidSynchronizationService {
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private EPersonService ePersonService;
+
+    @Override
+    public void linkProfile(Context context, Item profile, OrcidTokenResponseDTO token) throws SQLException {
+
+        String orcid = token.getOrcid();
+        String accessToken = token.getAccessToken();
+        String refreshToken = token.getRefreshToken();
+        String[] scopes = token.getScopeAsArray();
+
+        itemService.setMetadataSingleValue(context, profile, "person", "identifier", "orcid", null, orcid);
+        itemService.setMetadataSingleValue(context, profile, "dspace", "orcid", "access-token", null, accessToken);
+        itemService.setMetadataSingleValue(context, profile, "dspace", "orcid", "refresh-token", null, refreshToken);
+        itemService.clearMetadata(context, profile, "dspace", "orcid", "scope", Item.ANY);
+        for (String scope : scopes) {
+            itemService.addMetadata(context, profile, "dspace", "orcid", "scope", null, scope);
+        }
+
+        if (isBlank(itemService.getMetadataFirstValue(profile, "dspace", "orcid", "authenticated", Item.ANY))) {
+            String currentDate = ISO_DATE_TIME.format(now());
+            itemService.setMetadataSingleValue(context, profile, "dspace", "orcid", "authenticated", null, currentDate);
+        }
+
+        EPerson ePerson = ePersonService.findByProfileItem(context, profile);
+        EPerson ePersonByOrcid = ePersonService.findByNetid(context, orcid);
+        if (ePerson != null && ePersonByOrcid == null && isBlank(ePerson.getNetid())) {
+            ePerson.setNetid(orcid);
+            updateEPerson(context, ePerson);
+        }
+
+        updateItem(context, profile);
+
+    }
+
+    @Override
+    public void unlinkProfile(Context context, Item profile) throws SQLException {
+
+        itemService.clearMetadata(context, profile, "person", "identifier", "orcid", Item.ANY);
```

Comment:
Wondering if perhaps configuring a spring mapping between orcid token json fields and regular dspace `mdFields` wouldn't make [[this](https://github.com/4Science/DSpace/compare/CST-5306...4Science:CST-5669#diff-1faa212d2e33ccf408e44ff42446469680cfa386916ebe5ef16e2fe07f0bba16R67-R99)](https://github.com/4Science/DSpace/compare/CST-5306...4Science:CST-5669#diff-1faa212d2e33ccf408e44ff42446469680cfa386916ebe5ef16e2fe07f0bba16R67-R99) more customisable/easier to adapt to changei; but would likely also affect how these fields are used in other orcid PR's, since then you could have this single source of truth for all orcid related fields. Same for for example the mapping between eperson md fields and researchProfile md fields done [[here](https://github.com/4Science/DSpace/compare/CST-5306...4Science:CST-5669#diff-c3d89948e276dff86bced403ce9945fee9402a7c95228ea1d11dddba7a661b39R53)](https://github.com/4Science/DSpace/compare/CST-5306...4Science:CST-5669#diff-c3d89948e276dff86bced403ce9945fee9402a7c95228ea1d11dddba7a661b39R53). Just a lot of repeated explicit string md fields, would require code changes in a lot of places if any of the fields need changing/updating.